pkg/client/cache/reflector.go

```diff
@@ -215,3 +220,8 @@ loop:
 	glog.V(4).Infof("Watch close - %v total %v items received", r.expectedType, eventCount)
 	return nil
 }
+
+// LastSyncResourceVersion is the resource version observed when last sync with the underlying store
+func (r *Reflector) LastSyncResourceVersion() string {
+	return r.lastSyncResourceVersion
```

Comment:
Suggestion, maybe:

```go
func (r *Reflector) IsMostRecentResourceVersion(rv string) bool {
	return r.lastSyncResourceVersion == rv
}
```

will help people use this safely?

Also please comment on the thread safety here. I'm having a hard time convincing myself that there's a race-free way to use this.